File: packages/ttagy-client/examples/stream_chat.ts
import { TtagyClient } from "../src/index.js";

async function main() {
  const client = new TtagyClient({
    socketPath: "/tmp/ttagy.sock",
    autoFallback: true,
  });

  console.log("⚡ Starting streaming chat via TTAgy Client...");

  const stream = await client.streamChat({
    prompt: "Write a modern TypeScript discriminated union pattern",
    model: "gemini-3.7-flash",
    effort: "low",
  });

  for await (const event of stream) {
    if (event.type === "agy:thinking_delta") {
      process.stderr.write(`[Thinking] ${event.textDelta}`);
    } else if (event.type === "agy:content_delta") {
      process.stdout.write(event.textDelta);
    } else if (event.type === "agy:done") {
      console.log(`\n\n✅ Finished in ${event.elapsedMs}ms`);
    } else if (event.type === "agy:error") {
      console.error(`\n❌ Error: ${event.errorMessage}`);
    }
  }
}

main().catch(console.error);

File: crates/ttagy-client/examples/stream_chat.rs
use futures_util::StreamExt;
use ttagy_client::{ClientConfig, TtagyClient, TtagyRequest, TtagyStreamEvent};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = TtagyClient::new(ClientConfig::default());

    let req = TtagyRequest::builder("Write a high-performance concurrent queue in Rust")
        .model("gemini-3.7-flash")
        .effort("low")
        .build();

    let mut stream = client.stream_chat(req).await?;

    while let Some(event_res) = stream.next().await {
        match event_res? {
            TtagyStreamEvent::ThinkingDelta { text_delta, .. } => {
                eprint!("{}", text_delta);
            }
            TtagyStreamEvent::ContentDelta { text_delta, .. } => {
                print!("{}", text_delta);
            }
            TtagyStreamEvent::Done { elapsed_ms, .. } => {
                println!("\n\n✅ Done in {:.2}ms", elapsed_ms);
            }
            TtagyStreamEvent::Error { error_message, .. } => {
                eprintln!("\n❌ Error: {}", error_message);
            }
            _ => {}
        }
    }

    Ok(())
}
